lib/enum-validator.coffee

```diff
@@ -17,30 +17,27 @@ class EnumValidator
       generatedEnums: {}
 
   validateProduct: (product, resolvedProductType) =>
-    new Promise (resolve) =>
-      enumAttributes = @_fetchEnumAttributesFromProduct(product, resolvedProductType)
-      @_validateEnums(enumAttributes, resolvedProductType)
-      .then (updateActions) ->
-        update =
-          productTypeId: resolvedProductType.id
-          actions: updateActions
-        resolve(update)
+    enumAttributes = @_fetchEnumAttributesFromProduct(product, resolvedProductType)
+    updateActions = @_validateEnums(enumAttributes, resolvedProductType)
+    update =
+      productTypeId: resolvedProductType.id
+      actions: updateActions
+    update
 
   _validateEnums: (enumAttributes, productType) =>
-    new Promise (resolve, reject) =>
-      updateActions = []
-      referenceEnums = @_fetchEnumAttributesOfProductType(productType)
-      for ea in enumAttributes
-        if not @_isEnumGenerated(ea)
-          refEnum = _.findWhere(referenceEnums, {name: "#{ea.name}"})
-          if refEnum
-            if not @_isEnumKeyPresent(ea, refEnum)
-              updateActions.push @_generateUpdateAction(ea, refEnum)
-          else
-            reject("enum attribute name: #{ea.name} not found in Product Type: #{productType.name}", ea)
+    updateActions = []
+    referenceEnums = @_fetchEnumAttributesOfProductType(productType)
+    for ea in enumAttributes
+      if not @_isEnumGenerated(ea)
+        refEnum = _.findWhere(referenceEnums, {name: "#{ea.name}"})
+        if refEnum
+          if not @_isEnumKeyPresent(ea, refEnum)
+            updateActions.push @_generateUpdateAction(ea, refEnum)
         else
-          debug "Skipping #{ea.name} update action generation as already exists."
-      resolve(updateActions)
+          throw err "enum attribute name: #{ea.name} not found in Product Type: #{productType.name}", ea
```

Comment:
not really.. was rejecting earlier... but throw is not a replacement... made it into a debug statement